### ISSUE TITLE
[Proposal] Analyzing Humor Detection in LLMs

### ISSUE THREAD

**Issue by shanaj304** [uchicago-nlp-course/winter-2026-students#10]
[Project_proposal.pdf](https://github.com/user-attachments/files/25134922/Project_proposal.pdf)

**Comment by darinkeng** [comment]
Strengths:

- I like the idea of controlling possible confounders, and your idea of using aligned “unfunned” rewrites is a strong way to control for that :)
- It’s good that you understand the repo setup and you’re extending it from correlational evidence (linear probing and PCA) to causal evidence (ablation and steering). 

Weakness:

- GPT-2 Small is a pretty old and small model with poor reasoning capabilities. It likely does not really understand complex humor. I would also suggest trying newer models like Llama-3-8B or Gemma-2-9B. These models are small enough to run and more capable of catching "humor".
- Training SAEs on GPT2 is computationally heavy and can be a burden by putting it in the final week. I would suggest using pre-trained SAEs. There are also pretrained SAEs for Llama and Gemma models. I found this repo online and I hope this will help you: https://github.com/EleutherAI/sparsify

Next steps:

- You have done a great job on your week 6 progress. It's awesome that you're able to do this many experiments in one week :) The cosine similarity of 0.62 may suggest that Dataset B captures almost 40% noise/topic confounders rather than pure humor. You mentioned that steering inverts in the early layers, which might suggest that the Humor Direction in those layers is just syntax disruption.
- I hope that you can also try a newer model as another baseline if you have time. As for the SAE, it's fine that you can still test it on GPT2, but it would be better if you could extend it to newer models. Remember to report the results in your weekly blogs, either as a table or a figure plot. Great job!!

**Comment by shanaj304** [comment]
[Blog Post 1.pdf](https://github.com/user-attachments/files/25464865/Blog.Post.1.pdf)

**Comment by darinkeng** [comment]
Strengths:
- It’s great that you upgraded the GPT2 model to a more modern Gemma model
- Choosing to use a 1D projection for classification rather than a full logistic regression is a smart move. It forces the evaluation to rely solely on the target humor direction, and controls other confounding variable

Weakness:
- Although you showed removing the 1D representation at layer 15 would possibly remove the model’s ability to detect humor, I think you can make it more robust. Perhaps train a probe on a later layer(e.g layer 20). If ablating the humor vector also causes the probe accuracy at later layers to drop to near chance. Then we can have greater confidence in saying that the humor vector at layer 15 is indeed crucial!

Next step:
- I think your next steps are solid! Make sure to categorize the “Unknown” tokens in your top SAE features. Keep up the great work!

**Comment by shanaj304** [comment]
[Blog Post 2.pdf](https://github.com/user-attachments/files/25640250/Blog.Post.2.pdf)

**Comment by darinkeng** [comment]
- Overall you made substantial progress last week! I really like the downstream probe test idea. By continuously ablating the humor direction across layers 15 through 19 and then probing at layer 20, this rigorously proves that the downstream layers rely on that specific pathway.
- Extracting sparse features and manually labeling them is great, but you can also use LLM as a judge to make your labels more robust! You can feed the top 20 activating contexts for a specific SAE feature into an LLM like Claude 3.5 Sonnet and prompt it to blindly define the common concept.
- I think you next steps are pretty solid! Great work on the project :)

**Comment by shanaj304** [comment]
[Project_Report.pdf](https://github.com/user-attachments/files/25826661/Project_Report.pdf)
